### DISCUSSION TITLE
Help with encoder control surface

### DISCUSSION THREAD

**Discussion by Estraxtos** [tttapa/Control-Surface#858] [Q&A]
Hi, I am a noob in this world, but i need to connect two encoders to the control surface, but i don´t understand the code if you can help me i eternelly gratefull, pls

The code where i am working

#include <Encoder.h>
#include <Control_Surface.h>

USBMIDI_Interface midi;
 
int CLK1 = 2;
int DATA1 = 3;

int CLK2 = 4;
int DATA2 = 5;

// Instantiate a CCAbsoluteEncoder object
CCAbsoluteEncoder enc = {
  {2, 3},       // pins
  MIDI_CC::Pan, // MIDI address (CC number + optional channel)
  1, 
};
 
void setup() {
  Control_Surface.begin(); // Initialize Control Surface
}
 
void loop() {
  Control_Surface.loop(); // Update the Control Surface
}

**Comment by Trickster-git** [comment]
Hi, here you can find examples https://tttapa.github.io/Control-Surface-doc/Doxygen/examples.html

2 encoder example 

#include <[Control_Surface.h](https://tttapa.github.io/Control-Surface-doc/Doxygen/db/dbf/Control__Surface_8h.html)>
[USBMIDI_Interface](https://tttapa.github.io/Control-Surface-doc/Doxygen/de/d05/classUSBMIDI__Interface.html) midi;

  CCAbsoluteEncoder Enc1  =
 {{10, 11},  //it`s board pins
 {0 , CHANNEL_2, CABLE_1}, 1,};
  CCAbsoluteEncoder Enc2 = {{12, 13}, {1 , CHANNEL_2, CABLE_1}, 1,};

void setup() {
  [Control_Surface](https://tttapa.github.io/Control-Surface-doc/Doxygen/d9/d38/Control__Surface__Class_8cpp.html#af7a5d250297d82b2b29605dd7ac250df).[begin](https://tttapa.github.io/Control-Surface-doc/Doxygen/dc/d45/classControl__Surface__.html#ab0bdf5cca484fb2ba637c39384b27fb2)(); // Initialize Control Surface
}
 
void loop() {
  [Control_Surface](https://tttapa.github.io/Control-Surface-doc/Doxygen/d9/d38/Control__Surface__Class_8cpp.html#af7a5d250297d82b2b29605dd7ac250df).[loop](https://tttapa.github.io/Control-Surface-doc/Doxygen/dc/d45/classControl__Surface__.html#afe461d27b9c48d5921c00d521181f12f)(); // Update the Control Surface
}


which board are you use?

**Reply by Estraxtos** [reply]
Arduino Uno is the board

**Reply by Trickster-git** [reply]
if uno have ATmega16U2 you need use hairless midi serial converter http://projectgus.github.io/hairless-midiserial/?source=post_page---------------------------
and loopmidi 

after you finish your skech you can change 16u2 to usb midi device using flip.

**Reply by Estraxtos** [reply]
What about the button of the encoder? I have to put in:  pinMode(encbut1, INPUT_PULLUP);

and use this?

(this part is before void setup)

const int encbut1 = 6; / Button pin
int speedmultiply = 5;

long TimeOfLastDebounce = 0;
int DelayofDebounce = 0.01;


if ((millis() - TimeOfLastDebounce) > DelayofDebounce) {
    TimeOfLastDebounce = millis();
  }

          if (digitalRead(encbut1) == LOW) {  
    enc.speedMultiply  = 1;
  }

**Reply by Trickster-git** [reply]
not, you need button func, i share my code, you can find answer there 
[addPush-pico_2nd.zip](https://github.com/tttapa/Control-Surface/files/10133950/addPush-pico_2nd.zip)

**Reply by Estraxtos** [reply]
Thanks man!

**Reply by Trickster-git** [reply]
Many years ago, I have same questions.
